Electron 33 broken

The latest NAN release works fine in Electron 32,  but now fails with:

```
In file included from ../node_modules/nan/nan.h:3045:
../node_modules/nan/nan_scriptorigin.h:40:7: error: no matching constructor for initialization of 'v8::ScriptOrigin'
   40 |       v8::ScriptOrigin(v8::Isolate::GetCurrent()
      |       ^                ~~~~~~~~~~~~~~~~~~~~~~~~~
   41 |                    , name
      |                    ~~~~~~
   42 |                    , To<int32_t>(line).FromMaybe(0)) {}
../node_modules/nan/nan_scriptorigin.h:47:7: error: no matching constructor for initialization of 'v8::ScriptOrigin'
   47 |       v8::ScriptOrigin(v8::Isolate::GetCurrent()
      |       ^                ~~~~~~~~~~~~~~~~~~~~~~~~~
   48 |                    , name
      |                    ~~~~~~
   49 |                    , To<int32_t>(line).FromMaybe(0)
      |                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   50 |                    , To<int32_t>(column).FromMaybe(0)) {}

../node_modules/nan/nan_scriptorigin.h:36:7: error: no matching constructor for initialization of 'v8::ScriptOrigin'
   36 |       v8::ScriptOrigin(v8::Isolate::GetCurrent(), name) {}
      |       ^                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/qwe/Library/Caches/node-gyp/33.0.0/include/node/v8-message.h:64:13: note: candidate constructor not viable: no known conversion from 'Isolate *' to 'Local<Value>' for 1st argument
   64 |   V8_INLINE ScriptOrigin(Local<Value> resource_name,
      |             ^            ~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/qwe/Library/Caches/node-gyp/33.0.0/include/node/v8-message.h:62:17: note: candidate constructor (the implicit copy constructor) not viable: requires 1 argument, but 2 were provided
   62 | class V8_EXPORT ScriptOrigin {
      |                 ^~~~~~~~~~~~
/Users/qwe/Library/Caches/node-gyp/33.0.0/include/node/v8-message.h:62:17: note: candidate constructor (the implicit move constructor) not viable: requires 1 argument, but 2 were provided
   62 | class V8_EXPORT ScriptOrigin {
      |                 ^~~~~~~~~~~~
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -17,7 +17,7 @@ jobs:
     steps:
       - uses: actions/checkout@v4
       - uses: pre-commit/action@v3.0.1
-      
+
   ci:
     strategy:
       fail-fast: false
@@ -35,14 +35,6 @@ jobs:
             os: ubuntu-24.04-arm  # Linux on arm64
           - node-version: lts/*
             os: windows-2025
-          - node-version: 14.x
-            os: windows-2019
-          - node-version: 12.x
-            os: windows-2019
-          - node-version: 10.x
-            os: windows-2019
-          - node-version: 8.x
-            os: windows-2019
     runs-on: ${{ matrix.os }}
     steps:
       - uses: actions/checkout@v4

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -14,7 +14,8 @@ image:
 environment:
   NODE_GYP_FORCE_PYTHON: C:\Python39-x64\python.exe
   matrix:  # Test against these versions of Io.js and Node.js.
-    # - nodejs_version: "23"  # Enable after nodejs/nan#979 or similar.
+    - nodejs_version: "24"
+    - nodejs_version: "23"
     - nodejs_version: "22"
     - nodejs_version: "22"
       NODE_GYP_FORCE_PYTHON: C:\Python312-x64\python.exe

--- a/nan_callbacks.h
+++ b/nan_callbacks.h
@@ -13,6 +13,19 @@ template<typename T> class FunctionCallbackInfo;
 template<typename T> class PropertyCallbackInfo;
 template<typename T> class Global;
 
+#if defined(V8_MAJOR_VERSION) && (V8_MAJOR_VERSION > 12 ||                     \
+  (V8_MAJOR_VERSION == 12 && defined(V8_MINOR_VERSION) && V8_MINOR_VERSION > 4))
+namespace Intercepted {
+  constexpr v8::Intercepted No() { return v8::Intercepted::kNo; }
+  constexpr v8::Intercepted Yes() { return v8::Intercepted::kYes; }
+};
+#else
+namespace Intercepted {
+  inline void No() {}
+  inline void Yes() {}
+};
+#endif
+
 typedef void(*FunctionCallback)(const FunctionCallbackInfo<v8::Value>&);
 typedef void(*GetterCallback)
     (v8::Local<v8::String>, const PropertyCallbackInfo<v8::Value>&);

--- a/nan_callbacks_12_inl.h
+++ b/nan_callbacks_12_inl.h
@@ -111,7 +111,19 @@ class FunctionCallbackInfo {
   }
 #endif
   inline v8::Local<v8::Value> Data() const { return data_; }
-  inline v8::Local<v8::Object> Holder() const { return info_.Holder(); }
+  inline v8::Local<v8::Object> Holder() const {
+#if defined(V8_MAJOR_VERSION) &&                                               \
+    (V8_MAJOR_VERSION > 12 ||                                                  \
+     (V8_MAJOR_VERSION == 12 &&                                                \
+      (defined(V8_MINOR_VERSION) &&                                            \
+       (V8_MINOR_VERSION > 5 ||                                                \
+        (V8_MINOR_VERSION == 5 && defined(V8_BUILD_NUMBER) &&                  \
+         V8_BUILD_NUMBER >= 214)))))
+    return info_.This();
+#else
+    return info_.Holder();
+#endif
+  }
   inline bool IsConstructCall() const { return info_.IsConstructCall(); }
   inline int Length() const { return info_.Length(); }
   inline v8::Local<v8::Value> operator[](int i) const { return info_[i]; }

--- a/nan_callbacks_12_inl.h
+++ b/nan_callbacks_12_inl.h
@@ -106,7 +106,9 @@ class FunctionCallbackInfo {
   }
 
 #if NODE_MAJOR_VERSION < 10
-  inline v8::Local<v8::Function> Callee() const { return info_.Callee(); }
+  NAN_DEPRECATED inline v8::Local<v8::Function> Callee() const {
+    return info_.Callee();
+  }
 #endif
   inline v8::Local<v8::Value> Data() const { return data_; }
   inline v8::Local<v8::Object> Holder() const { return info_.Holder(); }

--- a/nan_callbacks_pre_12_inl.h
+++ b/nan_callbacks_pre_12_inl.h
@@ -140,7 +140,9 @@ class FunctionCallbackInfo {
     return args_.Callee();
   }
   inline v8::Local<v8::Value> Data() const { return data_; }
-  inline v8::Local<v8::Object> Holder() const { return args_.Holder(); }
+  inline v8::Local<v8::Object> Holder() const {
+    return args_.Holder();
+  }
   inline bool IsConstructCall() const { return args_.IsConstructCall(); }
   inline int Length() const { return args_.Length(); }
   inline v8::Local<v8::Value> operator[](int i) const { return args_[i]; }

--- a/nan_callbacks_pre_12_inl.h
+++ b/nan_callbacks_pre_12_inl.h
@@ -136,7 +136,9 @@ class FunctionCallbackInfo {
     return ReturnValue<T>(return_value_);
   }
 
-  inline v8::Local<v8::Function> Callee() const { return args_.Callee(); }
+  NAN_DEPRECATED inline v8::Local<v8::Function> Callee() const {
+    return args_.Callee();
+  }
   inline v8::Local<v8::Value> Data() const { return data_; }
   inline v8::Local<v8::Object> Holder() const { return args_.Holder(); }
   inline bool IsConstructCall() const { return args_.IsConstructCall(); }

--- a/nan_scriptorigin.h
+++ b/nan_scriptorigin.h
@@ -11,25 +11,28 @@
 
 class ScriptOrigin : public v8::ScriptOrigin {
  public:
-#if defined(V8_MAJOR_VERSION) && (V8_MAJOR_VERSION > 12 ||                      \
-  (V8_MAJOR_VERSION == 12 && (defined(V8_MINOR_VERSION) && (V8_MINOR_VERSION > 6\
-      || (V8_MINOR_VERSION == 6 && defined(V8_BUILD_NUMBER)                     \
-          && V8_BUILD_NUMBER >= 175)))))
+#if defined(V8_MAJOR_VERSION) &&                                               \
+    (V8_MAJOR_VERSION > 12 ||                                                  \
+     (V8_MAJOR_VERSION == 12 &&                                                \
+      (defined(V8_MINOR_VERSION) &&                                            \
+       (V8_MINOR_VERSION > 6 ||                                                \
+        (V8_MINOR_VERSION == 6 && defined(V8_BUILD_NUMBER) &&                  \
+         V8_BUILD_NUMBER >= 175)))))
   explicit ScriptOrigin(v8::Local<v8::Value> name) :
       v8::ScriptOrigin(name) {}
 
   ScriptOrigin(v8::Local<v8::Value> name
              , v8::Local<v8::Integer> line) :
       v8::ScriptOrigin(name
-                   , To<int32_t>(line).FromMaybe(0)) {}
+                     , To<int32_t>(line).FromMaybe(0)) {}
 
   ScriptOrigin(v8::Local<v8::Value> name
              , v8::Local<v8::Integer> line
              , v8::Local<v8::Integer> column) :
       v8::ScriptOrigin(name
-                   , To<int32_t>(line).FromMaybe(0)
-                   , To<int32_t>(column).FromMaybe(0)) {}
-#elif defined(V8_MAJOR_VERSION) && (V8_MAJOR_VERSION > 9 ||                      \
+                     , To<int32_t>(line).FromMaybe(0)
+                     , To<int32_t>(column).FromMaybe(0)) {}
+#elif defined(V8_MAJOR_VERSION) && (V8_MAJOR_VERSION > 9 ||                    \
   (V8_MAJOR_VERSION == 9 && (defined(V8_MINOR_VERSION) && (V8_MINOR_VERSION > 0\
       || (V8_MINOR_VERSION == 0 && defined(V8_BUILD_NUMBER)                    \
           && V8_BUILD_NUMBER >= 1)))))

--- a/test/cpp/indexedinterceptors.cpp
+++ b/test/cpp/indexedinterceptors.cpp
@@ -80,6 +80,7 @@ NAN_INDEX_GETTER(IndexedInterceptor::PropertyGetter) {
   } else {
     info.GetReturnValue().Set(Nan::New("bar").ToLocalChecked());
   }
+  return Intercepted::Yes();
 }
 
 NAN_INDEX_SETTER(IndexedInterceptor::PropertySetter) {
@@ -94,12 +95,14 @@ NAN_INDEX_SETTER(IndexedInterceptor::PropertySetter) {
   } else {
     info.GetReturnValue().Set(info.This());
   }
+  return Intercepted::Yes();
 }
 
 NAN_INDEX_ENUMERATOR(IndexedInterceptor::PropertyEnumerator) {
   v8::Local<v8::Array> arr = Nan::New<v8::Array>();
   Set(arr, 0, Nan::New(42));
   info.GetReturnValue().Set(arr);
+  return Intercepted::Yes();
 }
 
 NAN_INDEX_DELETER(IndexedInterceptor::PropertyDeleter) {
@@ -107,15 +110,19 @@ NAN_INDEX_DELETER(IndexedInterceptor::PropertyDeleter) {
     ObjectWrap::Unwrap<IndexedInterceptor>(info.Holder());
   std::strncpy(interceptor->buf, "goober", sizeof (interceptor->buf));
   info.GetReturnValue().Set(True());
+  return Intercepted::Yes();
 }
 
 NAN_INDEX_QUERY(IndexedInterceptor::PropertyQuery) {
   if (index == 1) {
     info.GetReturnValue().Set(Nan::New<v8::Integer>(v8::DontEnum));
+    return Intercepted::Yes();
   }
   if (index == 42) {
     info.GetReturnValue().Set(Nan::New(0));
+    return Intercepted::Yes();
   }
+  return Intercepted::No();
 }
 
 NODE_MODULE(indexedinterceptors, IndexedInterceptor::Init)

--- a/test/cpp/indexedinterceptors.cpp
+++ b/test/cpp/indexedinterceptors.cpp
@@ -122,7 +122,7 @@ NAN_INDEX_QUERY(IndexedInterceptor::PropertyQuery) {
     info.GetReturnValue().Set(Nan::New(0));
     return Intercepted::Yes();
   }
-  return Intercepted::No();
+  return Intercepted::Yes();
 }
 
 NODE_MODULE(indexedinterceptors, IndexedInterceptor::Init)

--- a/test/cpp/namedinterceptors.cpp
+++ b/test/cpp/namedinterceptors.cpp
@@ -80,6 +80,7 @@ NAN_PROPERTY_GETTER(NamedInterceptor::PropertyGetter) {
   } else {
     info.GetReturnValue().Set(Nan::New("bar").ToLocalChecked());
   }
+  return Intercepted::Yes();
 }
 
 NAN_PROPERTY_SETTER(NamedInterceptor::PropertySetter) {
@@ -94,6 +95,7 @@ NAN_PROPERTY_SETTER(NamedInterceptor::PropertySetter) {
   } else {
     info.GetReturnValue().Set(info.This());
   }
+  return Intercepted::Yes();
 }
 
 NAN_PROPERTY_ENUMERATOR(NamedInterceptor::PropertyEnumerator) {
@@ -107,16 +109,20 @@ NAN_PROPERTY_DELETER(NamedInterceptor::PropertyDeleter) {
     ObjectWrap::Unwrap<NamedInterceptor>(info.Holder());
   std::strncpy(interceptor->buf, "goober", sizeof (interceptor->buf));
   info.GetReturnValue().Set(True());
+  return Intercepted::Yes();
 }
 
 NAN_PROPERTY_QUERY(NamedInterceptor::PropertyQuery) {
   Nan::Utf8String s(property);
   if (!std::strcmp(*s, "thing")) {
-    return info.GetReturnValue().Set(Nan::New<v8::Integer>(v8::DontEnum));
+    info.GetReturnValue().Set(Nan::New<v8::Integer>(v8::DontEnum));
+    return Intercepted::Yes();
   }
   if (!std::strcmp(*s, "value")) {
-    return info.GetReturnValue().Set(Nan::New(0));
+    info.GetReturnValue().Set(Nan::New(0));
+    return Intercepted::Yes();
   }
+  return Intercepted::No();
 }
 
 NODE_MODULE(namedinterceptors, NamedInterceptor::Init)

--- a/test/cpp/namedinterceptors.cpp
+++ b/test/cpp/namedinterceptors.cpp
@@ -122,7 +122,7 @@ NAN_PROPERTY_QUERY(NamedInterceptor::PropertyQuery) {
     info.GetReturnValue().Set(Nan::New(0));
     return Intercepted::Yes();
   }
-  return Intercepted::No();
+  return Intercepted::Yes();
 }
 
 NODE_MODULE(namedinterceptors, NamedInterceptor::Init)

--- a/test/cpp/news.cpp
+++ b/test/cpp/news.cpp
@@ -114,10 +114,7 @@ NAN_METHOD(NewScript) {
 }
 
 NAN_METHOD(NewScript2) {
-  v8::ScriptOrigin origin(
-#if NODE_MODULE_VERSION >= NODE_18_0_MODULE_VERSION
-    info.GetIsolate(),
-#endif
+  ScriptOrigin origin(
     New<v8::String>("x").ToLocalChecked());
   v8::Local<UnboundScript> script =
       New<UnboundScript>(
@@ -135,10 +132,7 @@ NAN_METHOD(CompileScript) {
 }
 
 NAN_METHOD(CompileScript2) {
-  v8::ScriptOrigin origin(
-#if NODE_MODULE_VERSION >= NODE_18_0_MODULE_VERSION
-    info.GetIsolate(),
-#endif
+  ScriptOrigin origin(
     New<v8::String>("x").ToLocalChecked());
   v8::Local<BoundScript> script =
       CompileScript(New("2+4").ToLocalChecked(), origin).ToLocalChecked();

--- a/test/js/accessors-test.js
+++ b/test/js/accessors-test.js
@@ -11,7 +11,7 @@ const test     = require('tap').test
     , bindings = require('bindings')({ module_root: testRoot, bindings: 'accessors' });
 
 test('accessors', function (t) {
-  t.plan(7)
+  t.plan(9)
   var settergetter = bindings.create()
   t.equal(settergetter.prop1, 'this is property 1')
   t.ok(settergetter.prop2 === '')
@@ -28,5 +28,8 @@ test('accessors', function (t) {
   t.equal(derived.prop1, 'this is property 1')
   derived.prop2 = 'setting a new value'
   t.equal(derived.prop2, 'setting a new value')
-  t.equal(settergetter.prop2, 'setting a new value')
+  t.equal(settergetter.prop2, 'setting a value')
+  settergetter.prop2 = 'setting another value'
+  t.equal(settergetter.prop2, 'setting another value')
+  t.equal(derived.prop2, 'setting a new value')
 })

--- a/test/js/accessors-test.js
+++ b/test/js/accessors-test.js
@@ -9,27 +9,50 @@
 const test     = require('tap').test
     , testRoot = require('path').resolve(__dirname, '..')
     , bindings = require('bindings')({ module_root: testRoot, bindings: 'accessors' });
+const nodeVersion = parseInt(process.versions.node.split('.')[0]);
 
 test('accessors', function (t) {
-  t.plan(9)
   var settergetter = bindings.create()
-  t.equal(settergetter.prop1, 'this is property 1')
-  t.ok(settergetter.prop2 === '')
-  settergetter.prop2 = 'setting a value'
-  t.equal(settergetter.prop2, 'setting a value')
-  t.equal(settergetter.log(),
-    'New()\n' +
-    'Prop1:GETTER(this is property 1)\n' +
-    'Prop2:GETTER()\n' +
-    'Prop2:SETTER(setting a value)\n' +
-    'Prop2:GETTER(setting a value)\n'
-  )
   var derived = Object.create(settergetter)
-  t.equal(derived.prop1, 'this is property 1')
-  derived.prop2 = 'setting a new value'
-  t.equal(derived.prop2, 'setting a new value')
-  t.equal(settergetter.prop2, 'setting a value')
-  settergetter.prop2 = 'setting another value'
-  t.equal(settergetter.prop2, 'setting another value')
-  t.equal(derived.prop2, 'setting a new value')
+  if(nodeVersion > 22){
+    t.plan(9)
+    t.equal(settergetter.prop1, 'this is property 1')
+    t.ok(settergetter.prop2 === '')
+    settergetter.prop2 = 'setting a value'
+    t.equal(settergetter.prop2, 'setting a value')
+    t.equal(settergetter.log(),
+        'New()\n' +
+        'Prop1:GETTER(this is property 1)\n' +
+        'Prop2:GETTER()\n' +
+        'Prop2:SETTER(setting a value)\n' +
+        'Prop2:GETTER(setting a value)\n'
+    )
+    t.equal(derived.prop1, 'this is property 1')
+    derived.prop2 = 'setting a new value'
+    t.equal(derived.prop2, 'setting a new value')
+    t.equal(settergetter.prop2, 'setting a value')
+    settergetter.prop2 = 'setting another value'
+    t.equal(settergetter.prop2, 'setting another value')
+    t.equal(derived.prop2, 'setting a new value')
+
+  }
+  else{
+    t.plan(7)
+    t.equal(settergetter.prop1, 'this is property 1')
+    t.ok(settergetter.prop2 === '')
+    settergetter.prop2 = 'setting a value'
+    t.equal(settergetter.prop2, 'setting a value')
+    t.equal(settergetter.log(),
+        'New()\n' +
+        'Prop1:GETTER(this is property 1)\n' +
+        'Prop2:GETTER()\n' +
+        'Prop2:SETTER(setting a value)\n' +
+        'Prop2:GETTER(setting a value)\n'
+    )
+    t.equal(derived.prop1, 'this is property 1')
+    derived.prop2 = 'setting a new value'
+    t.equal(derived.prop2, 'setting a new value')
+    t.equal(settergetter.prop2, 'setting a new value')
+
+  }
 })

--- a/test/js/methodswithdata-test.js
+++ b/test/js/methodswithdata-test.js
@@ -9,6 +9,7 @@
 const test     = require('tap').test
     , testRoot = require('path').resolve(__dirname, '..')
     , bindings = require('bindings')({ module_root: testRoot, bindings: 'methodswithdata' })
+const nodeVersion = parseInt(process.versions.node.split('.')[0]);
 
 test('SetMethod with data', function (t) {
   t.plan(1);
@@ -16,25 +17,45 @@ test('SetMethod with data', function (t) {
 });
 
 test('accessors with data', function (t) {
-  t.plan(9)
-  var settergetter = bindings.create()
-  t.equal(settergetter.prop1, 'this is property 1')
-  t.ok(settergetter.prop2 === '')
-  settergetter.prop2 = 'setting a value'
-  t.equal(settergetter.prop2, 'setting a value')
-  t.equal(settergetter.log(),
-    'New()\n' +
-    'Prop1:GETTER(this is property 1)\n' +
-    'Prop2:GETTER()\n' +
-    'Prop2:SETTER(setting a value)\n' +
-    'Prop2:GETTER(setting a value)\n'
-  )
-  var derived = Object.create(settergetter)
-  t.equal(derived.prop1, 'this is property 1')
-  derived.prop2 = 'setting a new value'
-  t.equal(derived.prop2, 'setting a new value')
-  t.equal(settergetter.prop2, 'setting a value')
-  settergetter.prop2 = 'setting another value'
-  t.equal(settergetter.prop2, 'setting another value')
-  t.equal(derived.prop2, 'setting a new value')
+    var settergetter = bindings.create()
+    var derived = Object.create(settergetter)
+    if (nodeVersion > 22) {
+        t.plan(9)
+        t.equal(settergetter.prop1, 'this is property 1')
+        t.ok(settergetter.prop2 === '')
+        settergetter.prop2 = 'setting a value'
+        t.equal(settergetter.prop2, 'setting a value')
+        t.equal(settergetter.log(),
+            'New()\n' +
+            'Prop1:GETTER(this is property 1)\n' +
+            'Prop2:GETTER()\n' +
+            'Prop2:SETTER(setting a value)\n' +
+            'Prop2:GETTER(setting a value)\n'
+        )
+        t.equal(derived.prop1, 'this is property 1')
+        derived.prop2 = 'setting a new value'
+        t.equal(derived.prop2, 'setting a new value')
+        t.equal(settergetter.prop2, 'setting a value')
+        settergetter.prop2 = 'setting another value'
+        t.equal(settergetter.prop2, 'setting another value')
+        t.equal(derived.prop2, 'setting a new value')
+    } else {
+        t.plan(7)
+        t.equal(settergetter.prop1, 'this is property 1')
+        t.ok(settergetter.prop2 === '')
+        settergetter.prop2 = 'setting a value'
+        t.equal(settergetter.prop2, 'setting a value')
+        t.equal(settergetter.log(),
+            'New()\n' +
+            'Prop1:GETTER(this is property 1)\n' +
+            'Prop2:GETTER()\n' +
+            'Prop2:SETTER(setting a value)\n' +
+            'Prop2:GETTER(setting a value)\n'
+        )
+        t.equal(derived.prop1, 'this is property 1')
+        derived.prop2 = 'setting a new value'
+        t.equal(derived.prop2, 'setting a new value')
+        t.equal(settergetter.prop2, 'setting a new value')
+
+  }
 })

--- a/test/js/methodswithdata-test.js
+++ b/test/js/methodswithdata-test.js
@@ -11,28 +11,30 @@ const test     = require('tap').test
     , bindings = require('bindings')({ module_root: testRoot, bindings: 'methodswithdata' })
 
 test('SetMethod with data', function (t) {
-    t.plan(1);
-    t.ok(bindings.testWithData());
+  t.plan(1);
+  t.ok(bindings.testWithData());
 });
 
 test('accessors with data', function (t) {
-    t.plan(7)
-    var settergetter = bindings.create()
-    t.equal(settergetter.prop1, 'this is property 1')
-    t.ok(settergetter.prop2 === '')
-    settergetter.prop2 = 'setting a value'
-    t.equal(settergetter.prop2, 'setting a value')
-    t.equal(settergetter.log(),
-      'New()\n' +
-      'Prop1:GETTER(this is property 1)\n' +
-      'Prop2:GETTER()\n' +
-      'Prop2:SETTER(setting a value)\n' +
-      'Prop2:GETTER(setting a value)\n'
-    )
-    var derived = Object.create(settergetter)
-    t.equal(derived.prop1, 'this is property 1')
-    derived.prop2 = 'setting a new value'
-    t.equal(derived.prop2, 'setting a new value')
-    t.equal(settergetter.prop2, 'setting a new value')
-  })
-  
+  t.plan(9)
+  var settergetter = bindings.create()
+  t.equal(settergetter.prop1, 'this is property 1')
+  t.ok(settergetter.prop2 === '')
+  settergetter.prop2 = 'setting a value'
+  t.equal(settergetter.prop2, 'setting a value')
+  t.equal(settergetter.log(),
+    'New()\n' +
+    'Prop1:GETTER(this is property 1)\n' +
+    'Prop2:GETTER()\n' +
+    'Prop2:SETTER(setting a value)\n' +
+    'Prop2:GETTER(setting a value)\n'
+  )
+  var derived = Object.create(settergetter)
+  t.equal(derived.prop1, 'this is property 1')
+  derived.prop2 = 'setting a new value'
+  t.equal(derived.prop2, 'setting a new value')
+  t.equal(settergetter.prop2, 'setting a value')
+  settergetter.prop2 = 'setting another value'
+  t.equal(settergetter.prop2, 'setting another value')
+  t.equal(derived.prop2, 'setting a new value')
+})

--- a/test/js/weak-test.js
+++ b/test/js/weak-test.js
@@ -10,38 +10,46 @@ const test     = require('tap').test
     , testRoot = require('path').resolve(__dirname, '..')
     , bindings = require('bindings')({ module_root: testRoot, bindings: 'weak' })
     , gc = require('./gc-fn');
+const nodeVersion = parseInt(process.versions.node.split('.')[0]);
 
-test('weak', function (t) {
-  t.plan(3);
+// https://github.com/nodejs/nan/issues/995
+// https://chromium-review.googlesource.com/c/v8/v8/+/4733273
+// Week tests are incompatible with recent V8 changes
 
-  var weak = bindings, count = 0;
-  t.type(weak.hustle, 'function');
+if(nodeVersion < 24){
+  test('weak', function (t) {
+    t.plan(3);
 
-  weak.hustle(function () {}, function (val) {
-    t.equal(val, 42);
-    count++;
+    var weak = bindings, count = 0;
+    t.type(weak.hustle, 'function');
+
+    weak.hustle(function () {}, function (val) {
+      t.equal(val, 42);
+      count++;
+    });
+
+    var timeout = setTimeout(function () {
+      // run weak callback, should dispose
+      gc();
+
+      // do not run weak callback
+      gc();
+
+      if (count > 0) {
+        clearTimeout(timeout);
+        t.equal(count, 1);
+      }
+    }, 100);
   });
 
-  var timeout = setTimeout(function () {
-    // run weak callback, should dispose
-    gc();
+  test('weak external', function (t) {
+    t.plan(2);
 
-    // do not run weak callback
-    gc();
+    var weak = bindings;
+    t.type(weak.weakExternal, 'function');
 
-    if (count > 0) {
-      clearTimeout(timeout);
-      t.equal(count, 1);
-    }
-  }, 100);
-});
+    weak.weakExternal();
+    t.ok(true);  // All good if the previous line didn't crash.
+  });
+}
 
-test('weak external', function (t) {
-  t.plan(2);
-
-  var weak = bindings;
-  t.type(weak.weakExternal, 'function');
-
-  weak.weakExternal();
-  t.ok(true);  // All good if the previous line didn't crash.
-});

--- a/test/js/weak2-test.js
+++ b/test/js/weak2-test.js
@@ -10,28 +10,35 @@ const test     = require('tap').test
     , testRoot = require('path').resolve(__dirname, '..')
     , bindings = require('bindings')({ module_root: testRoot, bindings: 'weak2' })
     , gc = require('./gc-fn');
+const nodeVersion = parseInt(process.versions.node.split('.')[0]);
 
-test('weak2', function (t) {
-  t.plan(3);
+// https://github.com/nodejs/nan/issues/995
+// https://chromium-review.googlesource.com/c/v8/v8/+/4733273
+// Week tests are incompatible with recent V8 changes
 
-  var weak2 = bindings, count = 0;
-  t.type(weak2.hustle, 'function');
+if(nodeVersion < 24){
+  test('weak2', function (t) {
+    t.plan(3);
 
-  weak2.hustle(function (val) {
-    t.equal(val, 42);
-    count++;
+    var weak2 = bindings, count = 0;
+    t.type(weak2.hustle, 'function');
+
+    weak2.hustle(function (val) {
+      t.equal(val, 42);
+      count++;
+    });
+
+    // run weak callback, should dispose
+    gc();
+
+    // do not run weak callback
+    gc();
+
+    var timeout = setTimeout(function () {
+      if (count > 0) {
+        clearTimeout(timeout);
+        t.equal(count, 1);
+      }
+    }, 100);
   });
-
-  // run weak callback, should dispose
-  gc();
-
-  // do not run weak callback
-  gc();
-
-  var timeout = setTimeout(function () {
-    if (count > 0) {
-      clearTimeout(timeout);
-      t.equal(count, 1);
-    }
-  }, 100);
-});
+}

